cmd/terramate/cli/run_script.go

```diff
@@ -0,0 +1,168 @@
+	}
+
+	var stacks config.List[*config.SortableStack]
+	if c.parsedArgs.Experimental.RunScript.NoRecursive {
+		st, found, err := config.TryLoadStack(c.cfg(), prj.PrjAbsPath(c.rootdir(), c.wd()))
+		if err != nil {
+			logger.Fatal().Msgf("failed to load stack in current directory: %s", err)
+		}
+
+		if !found {
+			logger.Fatal().Msg("--no-recursive provided but no stack found in the current directory")
+		}
+
+		stacks = append(stacks, st.Sortable())
+	} else {
+		var err error
+		stacks, err = c.computeSelectedStacks(true)
+		if err != nil {
+			logger.Fatal().Msgf("failed to compute selected stacks %s", err)
+		}
+	}
+
+	// search for the script and prepare a list of script/stack entries
+	m := newScriptsMatcher(c.parsedArgs.Experimental.RunScript.Labels)
+	m.Search(c.cfg().Tree(), &stacks)
+
+	if len(m.Results) == 0 {
+		c.output.MsgStdErr(color.RedString("script not found: ") +
+			strings.Join(c.parsedArgs.Experimental.RunScript.Labels, " "))
+		os.Exit(1)
+	}
+
+	for _, result := range m.Results {
+		c.output.MsgStdErr("Found %s defined at %s having %s job(s)",
+			color.GreenString(result.ScriptCfg.Name()),
+			color.BlueString(result.ScriptCfg.Range.String()),
+			color.BlueString(fmt.Sprintf("%d", len(result.ScriptCfg.Jobs))),
+		)
+
+		if len(result.Stacks) == 0 {
+			c.output.MsgStdOut(color.RedString("(skipping) "),
+				"provided criteria does not allow script execution",
+			)
+		}
+
+		for _, st := range result.Stacks {
+			ectx, err := scriptEvalContext(c.cfg(), st.Stack)
+			if err != nil {
+				logger.Fatal().Msgf("failed to get context: %s", err)
+			}
+
+			evalScript, err := config.EvalScript(ectx, *result.ScriptCfg)
+			if err != nil {
+				logger.Fatal().Msgf("failed to eval script: %s", err)
+			}
+
+			for jobNum, j := range evalScript.Jobs {
+				c.output.MsgStdOut("")
+				for cmdNum, cmd := range j.Commands() {
+					printCommand(c.stderr, cmd, st.Dir().String(), jobNum, cmdNum)
+
+					env, err := run.LoadEnv(c.cfg(), st.Stack)
+					if err != nil {
+						logger.Fatal().Msgf("failed to load env: %s", err)
+					}
+
+					if err := c.executeCommand(
+						cmd, st.Dir().HostPath(c.rootdir()), env, c.parsedArgs.Experimental.RunScript.DryRun,
+					); err != nil {
+						logger.Fatal().Msgf("unable to execute command: %s", err)
+					}
+					c.output.MsgStdOut("")
+				}
+			}
+		}
+	}
+}
+
+func (c *cli) executeCommand(cmd []string, wd string, env []string, dryRun bool) error {
+	if dryRun {
+		c.output.MsgStdOut(color.BlueString("# -- this is a dry run -- #"))
+		return nil
+	}
+
+	newCmd, err := makeCommand(cmd, wd, env, c.stdout, c.stderr)
+	if err != nil {
+		return fmt.Errorf("failed to prepare command: %s", err)
+	}
+
+	if err := newCmd.Run(); err != nil {
+		return fmt.Errorf("failed to execute command: %s", err)
+	}
+
+	return nil
+}
+
+func makeCommand(command []string, dir string, env []string, stdout, stderr io.Writer) (*exec.Cmd, error) {
+	cmdPath, err := exec.LookPath(command[0])
```

Comment:
This must use the `run.LookPath()` so it uses the provided environ to lookup the path.

Reply:
Fixed